src/WPF/ArcGISRuntime.WPF.Samples/Samples/Data/EditAndSyncFeatures/EditAndSyncFeatures.xaml.cs

```diff
@@ -0,0 +1,511 @@
+                double tolerance = 15 * MyMapView.UnitsPerPixel;
+
+                // Define the selection envelope
+                Envelope selectionEnvelope = new Envelope(e.Location.X - tolerance, e.Location.Y - tolerance, e.Location.X + tolerance, e.Location.Y + tolerance);
+
+                // Define query parameters for feature selection
+                QueryParameters query = new QueryParameters()
+                {
+                    Geometry = selectionEnvelope
+                };
+
+                // Select the feature in all applicable tables
+                foreach (FeatureLayer layer in MyMapView.Map.OperationalLayers)
+                {
+                    await layer.SelectFeaturesAsync(query, SelectionMode.New);
+                }
+
+                // Set the edit state
+                _readyForEdits = EditState.Editing;
+            }
+        }
+
+        private void UpdateMapExtent()
+        {
+            // Return if mapview is null
+            if (MyMapView == null) { return; }
+
+            // Get the new viewpoint
+            Viewpoint myViewPoint = MyMapView.GetCurrentViewpoint(ViewpointType.BoundingGeometry);
+
+            // Return if viewpoint is null
+            if (myViewPoint == null) { return; }
+
+            // Get the updated extent for the new viewpoint
+            Envelope extent = myViewPoint.TargetGeometry as Envelope;
+
+            // Return if extent is null
+            if (extent == null) { return; }
+
+            // Create an envelope that is a bit smaller than the extent
+            EnvelopeBuilder envelopeBldr = new EnvelopeBuilder(extent);
+            envelopeBldr.Expand(0.80);
+
+            // Get the (only) graphics overlay in the map view
+            var extentOverlay = MyMapView.GraphicsOverlays.FirstOrDefault();
+
+            // Return if the extent overlay is null
+            if (extentOverlay == null) { return; }
+
+            // Get the extent graphic
+            Graphic extentGraphic = extentOverlay.Graphics.FirstOrDefault();
+
+            // Create the extent graphic and add it to the overlay if it doesn't exist
+            if (extentGraphic == null)
+            {
+                extentGraphic = new Graphic(envelopeBldr.ToGeometry());
+                extentOverlay.Graphics.Add(extentGraphic);
+            }
+            else
+            {
+                // Otherwise, simply update the graphic's geometry
+                extentGraphic.Geometry = envelopeBldr.ToGeometry();
+            }
+        }
+
+        private async void StartGeodatabaseGeneration()
+        {
+            // Create a task for generating a geodatabase (GeodatabaseSyncTask)
+            _gdbSyncTask = await GeodatabaseSyncTask.CreateAsync(_featureServiceUri);
+
+            // Get the current extent of the red preview box
+            Envelope extent = MyMapView.GraphicsOverlays.FirstOrDefault().Extent as Envelope;
```

Comment:
This is like a coding "run-on sentence". Break the logic into 2 lines of code: one for getting the graphic and another for getting the Extent.